diff -rupN mingw-w64-v6.0.0/mingw-w64-tools/widl/include/windef.h mingw-w64-v6.0.0-new/mingw-w64-tools/widl/include/windef.h
--- mingw-w64-v6.0.0/mingw-w64-tools/widl/include/windef.h	2018-09-14 16:16:43.000000000 +0200
+++ mingw-w64-v6.0.0-new/mingw-w64-tools/widl/include/windef.h	2019-06-04 17:41:05.156136224 +0200
@@ -101,7 +101,7 @@ extern "C" {
 #endif /* __cdecl */
 
 #ifndef __ms_va_list
-# if (defined(__x86_64__) || defined(__aarch64__)) && defined (__GNUC__)
+# if (defined(__x86_64__)) && defined (__GNUC__)
 #  define __ms_va_list __builtin_ms_va_list
 #  define __ms_va_start(list,arg) __builtin_ms_va_start(list,arg)
 #  define __ms_va_end(list) __builtin_ms_va_end(list)
